wip-0004.md

```diff
@@ -0,0 +1,166 @@
+
+                            Block Header
+
+    Data request root       Tally root     Rest of the transactions root 
+
+In order to construct a **superblock**, we propose the following schema:
+
+- Nodes compute an ordered merkle root of all block all data request roots that were consolidated in the last *checkpoint* epochs.
+
+- Nodes compute an ordered merkle root of all block all tally roots that were consolidated in the last *checkpoint* epochs.
+
+- Nodes compute a merkle tree with the ARS snapshot at the time of proposing the superblock. This will be ordered by their public keys.
+
+```
+
+                                          Superblock
+
+        superblock_dr_root            superblock_tally_root               ARS_root
+
+      dr_root1       dr_root2       tally_root1       tally_root2     pk1          pk2 
+
+```
+
+Note that verifying data request and tally transactions becomes as simple as providing the valid merkle path to the superblock_dr_root/superblock_tally_root transactions. Identically, proving the membership to the ARS is achieved by again providing a valid merkle path to the ARS_root.
+
+**Specification**
+
+The message to be hashed and signed by the ARS members could look something like this:
+
+```rust
+struct Superblock {
+    struct BlockHeader {
+        struct BlockMerkleRoots {
+            dr_hash_merkle_root: Hash,      
+            tally_hash_merkle_root: Hash,
+            ars_merkle_root: Hash
+        }
+        superblock_beacon: CheckpointBeacon;
+    }
+}
+```
+where *dr_hash_merkle_root* and *tally_hash_merkle_root* are the merkle roots with all dr/tally roots from all blocks that occurred in the last *checkpoints. *ars_merkle_root* refers to the merkle root of the ARS members at the current checkpoint, and *superblock_beacon* refers to the index and the hash of the previous superblock.
+
+The message to be transmitted will be the signature of the hash of the aforementiond superblock:
+
+```rust
+message SuperblockVote {
+    KeyedSignature superblock_sig = 1;
+}
+```
+
+#### Signature aggregation
+The aggregation can happen at two different stages in the protocol:
+
+1. The ARS members append their signature to an aggregated signature they receive. The drawback of this option is that ARS members should verify the aggregated signatures to prevent from aggregating their own signature to an invalid set of signatures.
```

Comment:
```suggestion
1. The ARS members append their signature to an aggregated signature they receive. The drawback of this option is that ARS members should verify the aggregated signatures to prevent from adding their own signature to an invalid set of signatures.
```